common-content/en/module/complexity/memory-consumption/index.md

```diff
@@ -34,11 +34,14 @@ const userRoles = ["Admin", "Editor", "Viewer"]; //An array of 3 short strings
 const userProfiles = [ {id: 1, name: "Farzaneh", role: "Admin", preferences: {...}}, {id: 2, name: "Cuneyt", role: "Editor", preferences: {...}} ]; // An array of 2 complex objects
 ```
 
-Different kinds of data have different memory footprints:
+Different kinds of data have different memory footprints. All data if fundamentally stored as bytes. We can form intuition for how much memory a piece of data takes:
 
-- Numbers or booleans use less memory than objects
+- Numbers are typically stored as 8 bytes. In some languages, you can define numbers which take up less space (but can store a smaller range of values).
+- Each character in an ASCII string takes 1 byte. More complex characters may take more bytes. The biggest characters take up to 4 bytes.
 - The longer a string, the more bytes it consumes.
-- Objects and arrays need memory for their internal organisation _as well_ as the data itself.
+- Objects and arrays are stored in different ways in different languages. But they need to store _at least_ the information contained within them.
+  - This means an array of 5 elements will use _at least_ as much memory as the 5 elements would on their own.
+  - And objects will use _at least_ as much memory as all of the _values_ inside the object (and in some languages, all of the keys as well).
```

Comment:
Again, is this a bit too much info? We don't want to overload them.

Reply:
Maybe... Let's maybe have a chat about it?

I changed it because the previous content is kind of misleading - in most compiled programming languages, arrays and objects take exactly as much space as their contents/members/fields, so I wanted to avoid instilling an intuition that they have overhead where they don't necessarily...

Reply:
Hmm, it's a good point.  I was trying to steer away from the python size of debacle I've encountered before with trainees and have oversteered.